tools/topology/sof-cml-rt5682.m4

```diff
@@ -15,28 +15,11 @@ include(`common/tlv.m4')
 include(`sof/tokens.m4')
 
 # Include Cannonlake or Icelake DSP configuration
-ifelse(PLATFORM, `icl', include(`platform/intel/icl.m4'),
-	ifelse(PLATFORM, `whl', include(`platform/intel/cnl.m4'),
-		ifelse(PLATFORM, `cml', include(`platform/intel/cnl.m4'), `')))
+include(`platform/intel/all.m4')
 include(`platform/intel/dmic.m4')
 
 DEBUG_START
 
-undefine(`SSP_INDEX')
-define(`SSP_INDEX', ifelse(PLATFORM, `icl', `0',
-	ifelse(PLATFORM, `whl', `1',
-		ifelse(PLATFORM, `cml', `0', `'))))
-
-undefine(`SSP_NAME')
-define(`SSP_NAME', ifelse(PLATFORM, `icl', `SSP0-Codec',
-	ifelse(PLATFORM, `whl', `SSP1-Codec',
-		ifelse(PLATFORM, `cml', `SSP0-Codec', `'))))
-
-undefine(`SSP_MCLK_RATE')
-define(`SSP_MCLK_RATE', ifelse(PLATFORM, `icl', `19200000',
-	ifelse(PLATFORM, `whl', `24000000',
-		ifelse(PLATFORM, `cml', `24000000', `'))))
-
```

Comment:
sorry, it makes no sense to move this to a common code. This can only be used for cml-rt5682 based topology. The SSP_INDEX is really tied to a specific hw, it can't be reused blindly. NAK on this.

Reply:
Thanks for your comment. I will review again, any way I can make this simple and generic. I will think about belwo comments as well.